src/test/java/org/apache/sysds/test/functions/federated/primitives/FederatedFullCumulativeTest.java

```diff
@@ -0,0 +1,217 @@
+import org.apache.sysds.runtime.meta.MatrixCharacteristics;
+import org.apache.sysds.runtime.util.HDFSTool;
+import org.apache.sysds.test.AutomatedTestBase;
+import org.apache.sysds.test.TestConfiguration;
+import org.apache.sysds.test.TestUtils;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(value = Parameterized.class)
+@net.jcip.annotations.NotThreadSafe
+public class FederatedFullCumulativeTest extends AutomatedTestBase {
+	private final static String TEST_NAME1 = "FederatedCumsumTest";
+	private final static String TEST_NAME2 = "FederatedCumprodTest";
+	private final static String TEST_NAME3 = "FederatedCummaxTest";
+	private final static String TEST_NAME4 = "FederatedCumminTest";
+	private final static String TEST_NAME5 = "FederatedCumsumprodTest";
+
+	private final static String TEST_DIR = "functions/federated/cumulative/";
+	private static final String TEST_CLASS_DIR = TEST_DIR + FederatedFullCumulativeTest.class.getSimpleName() + "/";
+
+	private final static int blocksize = 1024;
+	@Parameterized.Parameter()
+	public int rows;
+	@Parameterized.Parameter(1)
+	public int cols;
+	@Parameterized.Parameter(2)
+	public boolean rowPartitioned;
+
+	@Parameterized.Parameters
+	public static Collection<Object[]> data() {
+		return Arrays.asList(
+			new Object[][] {
+				{24, 4, false},
+//				{24, 24, false},
+//				{1000, 4, false},
+//				{1000, 4, true},
+			});
+	}
+
+	private enum OpType {
+		SUM, PROD, SUMPROD, MAX, MIN
+	}
+
+	@Override
+	public void setUp() {
+		TestUtils.clearAssertionInformation();
+		addTestConfiguration(TEST_NAME1, new TestConfiguration(TEST_CLASS_DIR, TEST_NAME1, new String[] {"S"}));
+		addTestConfiguration(TEST_NAME2, new TestConfiguration(TEST_CLASS_DIR, TEST_NAME2, new String[] {"S"}));
+		addTestConfiguration(TEST_NAME3, new TestConfiguration(TEST_CLASS_DIR, TEST_NAME3, new String[] {"S"}));
+		addTestConfiguration(TEST_NAME4, new TestConfiguration(TEST_CLASS_DIR, TEST_NAME4, new String[] {"S"}));
+		addTestConfiguration(TEST_NAME5, new TestConfiguration(TEST_CLASS_DIR, TEST_NAME5, new String[] {"S"}));
+	}
+
+	@Test
+	public void testSumDenseMatrixCP() { runCumOperationTest(OpType.SUM, ExecType.CP); }
+
+	@Test
+	public void testProdDenseMatrixCP() { runCumOperationTest(OpType.PROD, ExecType.CP); }
+
+	@Test
+	public void testMaxDenseMatrixCP() {
+		runCumOperationTest(OpType.MAX, ExecType.CP);
+	}
+
+	@Test
+	public void testMinDenseMatrixCP() {
+		runCumOperationTest(OpType.MIN, ExecType.CP);
+	}
+
+	@Test
+	public void testSumprodDenseMatrixCP() {
+		runCumOperationTest(OpType.SUMPROD, ExecType.CP);
+	}
+
+	private void runCumOperationTest(OpType type, ExecType instType) {
+		ExecMode platformOld = rtplatform;
+		if(instType == ExecType.SPARK)
+			rtplatform = ExecMode.SPARK;
+		else
+			rtplatform = ExecMode.HYBRID;
+
+		if(rtplatform == ExecMode.SPARK)
+			DMLScript.USE_LOCAL_SPARK_CONFIG = true;
```

Comment:
Instead of this setup code, always use setExecMode, which helps avoid this redundant code in every test.